tools/wptrunner/wptrunner/executors/executorwebdriver.py

```diff
@@ -355,25 +355,27 @@ class WebDriverVirtualSensor(VirtualSensorProtocolPart):
     def setup(self):
         self.webdriver = self.parent.webdriver
 
-    def create_virtual_sensor(self, create_parameters):
-        print("JV666 > tools/wptrunner/wptrunner/executors/executorwebdriver.py create_virtual_sensor()\n")
-        print(create_parameters)
-        return self.webdriver.send_session_command("POST", "sensor", create_parameters)
-
-    def update_virtual_sensor(self, update_parameters):
-        print("JV666 > tools/wptrunner/wptrunner/executors/executorwebdriver.py update_virtual_sensor()\n")
-        type = update_parameters["type"]
-        return self.webdriver.send_session_command("POST", "sensor/%s" % type, update_parameters)
-
-    def remove_virtual_sensor(self, remove_parameters):
-        print("JV666 > tools/wptrunner/wptrunner/executors/executorwebdriver.py remove_virtual_sensor()\n")
-        type = remove_parameters["type"]
-        return self.webdriver.send_session_command("DELETE", "sensor/%s" % type, remove_parameters)
-
-    def get_virtual_sensor_information(self, information_parameters):
-        print("JV666 > tools/wptrunner/wptrunner/executors/executorwebdriver.py get_virtual_sensor_information()\n")
-        type = information_parameters["type"]
-        return self.webdriver.send_session_command("GET", "sensor/%s" % type, information_parameters)
+    def create_virtual_sensor(self, sensor_type, sensor_params):
+        if isinstance(sensor_params, dict):
```

Comment:
Thinking about this again, I believe the most idiomatic solution is to make sensor_params an optional parameter in the JS bits that defaults to `{}` (an Object) and always assume it's a dict here in Python. You could do something like this:

```js
// testdriver.js
... = create_virtual_sensor(sensor_type, sensor_params={}, context=null) {
  ...
}

// testdriver-extra.js
window.test_driver_internal.create_virtual_sensor = function(sensor_type, sensor_params={}, context=null) {
  ...
}
```

```python
# executorwebdriver.py
def create_virtual_sensor(self, sensor_type, sensor_params):
  body = {"type": sensor_type}
  body.update(sensor_params)
  return self.webddriver.send_session_command(...)
```

Reply:
Fixed in https://github.com/JuhaVainio/WPT_sensors/pull/1/commits/83cc4f75ef04a2699fd59f5ddfe0087a5a509369